engine/baml-lib/baml-core/src/ir/repr.rs

```diff
@@ -248,6 +248,58 @@ impl IntermediateRepr {
 
         Ok(repr)
     }
+
+    /// TODO: #1343 Temporary solution until we implement scoping in the AST.
```

Comment:
What is the reference to #1343 about? #1343 is the merged PR adding tests for dynamic types.